api/routes/ws/chat.dart

```diff
@@ -0,0 +1,48 @@
+import 'dart:convert';
+import 'package:api/chat/chat_bloc.dart';
+import 'package:api/chat/chat_model.dart';
+import 'package:dart_frog/dart_frog.dart';
+import 'package:dart_frog_web_socket/dart_frog_web_socket.dart';
+
+/// Websocket Handler
+Future<Response> onRequest(RequestContext context) async {
+  final handler = webSocketHandler((channel, protocol) {
+    final chatCubit = context.read<ChatCubit>()..subscribe(channel);
+
+    channel.sink.add(chatCubit.state.toString());
+    channel.stream.listen(
+      (data) {
+        try {
+          if (data == null || data.toString().isEmpty) {
+            channel.sink.add(
+              jsonEncode({
+                'status': 'Error invalid data:$data',
+              }),
+            );
+
+            return;
+          }
+          final jsonData = jsonDecode(data.toString());
+          if (jsonData is! Map<String, dynamic>) {
+            channel.sink.add(
+              jsonEncode({
+                'status': 'Error invalid data:$data',
+              }),
+            );
+            return;
+          }
+          final chatModel = ChatModel.fromMap(jsonData);
+          chatCubit.addToChat(chatModel);
+        } catch (e) {
+          channel.sink.add(
+            jsonEncode({'status': 'error', 'message': e.toString()}),
+          );
+        }
+      },
+      onDone: () {
+        chatCubit.unsubscribe(channel);
+      },
+    );
+  });
+  return handler(context);
```

Comment:
We can do all this in our original ws connection, we just have to handle type of message added in the channel sink. Eg:  __add_player__, __update_offsets__, __send_message__

Reply:
@gauravkatvaa and I were going to do this first
But wouldnt we risk performance by using 1 socket for everything?

Reply:
Performance won't be affected and multiple ws connections can cause synchronization problems. It's better to use a single connection throughout the game. We can differentiate events like I mentioned above

Reply:
right we can create an enum and on that basis we can pass the data to the stream